src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1491,6 +1492,67 @@ public function testParseMultiLineMappingValue()
 
         $this->assertEquals($expected, $this->parser->parse($yaml));
     }
+
+    public function testTaggedInlineMapping()
+    {
+        $this->assertEquals(new TaggedValue(array('foo' => 'bar'), 'foo'), $this->parser->parse('!foo {foo: bar}', Yaml::PARSE_CUSTOM_TAGS));
+    }
+
+    /**
+     * @dataProvider taggedValuesProvider
+     */
+    public function testCustomTagSupport($expected, $yaml)
+    {
+        $this->assertEquals($expected, $this->parser->parse($yaml, Yaml::PARSE_CUSTOM_TAGS));
+    }
+
+    public function taggedValuesProvider()
+    {
+        return array(
+            'sequences' => array(
+                array(new TaggedValue(array('yaml'), 'foo'), new TaggedValue(array('bar'), '!quz')),
+                <<<YAML
+- !foo
+    - yaml
+- !!quz [bar]
```

Comment:
Tags starting with two exclamation marks are tags defined by the YAML spec. I do not think we should support them through a flag that indicates to parse *custom* tags (and if we want to parse built-in tags, we will have to validate their names IMO).